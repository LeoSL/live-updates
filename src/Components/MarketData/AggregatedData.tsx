import { ComponentType } from "react";
import { Box, Table, TableBody, TableRow, TableCell } from "@material-ui/core";
import { OrdersTable } from "./OrdersTable";
import { Orderbook } from "../../models/pairs";

type AggregatedDataProps = {
  average: string;
  close: string;
  high: string;
  low: string;
  orderbook: Orderbook;
};

export const AggregatedData: ComponentType<AggregatedDataProps> = ({
  average,
  close,
  high,
  low,
  orderbook,
}) => {
  return (
    <Box display="flex" flexDirection="column">
      <Table>
        <TableBody>
          <TableRow>
            <TableCell>
              <strong>Average:</strong> {average}
            </TableCell>
            <TableCell>
              <strong>Close:</strong> {close}
            </TableCell>
            <TableCell>
              <strong>High:</strong> {high}
            </TableCell>
            <TableCell>
              <strong>Low:</strong> {low}
            </TableCell>
          </TableRow>
        </TableBody>
      </Table>
      <Box>
        <OrdersTable {...orderbook} />
      </Box>
    </Box>
  );
};
