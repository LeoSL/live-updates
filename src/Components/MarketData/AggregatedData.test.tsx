import { render } from "@testing-library/react";
import { AggregatedData } from "./AggregatedData";

it("renders correctly", () => {
  const orderbook = {
    asks: [{ price: "12000" }],
    bids: [{ price: "20000" }],
  };

  const props = {
    average: "1200",
    close: "1201",
    high: "1202",
    low: "1203",
    orderbook,
  };

  const { container } = render(<AggregatedData {...props} />);
  expect(container).toMatchSnapshot();
});
