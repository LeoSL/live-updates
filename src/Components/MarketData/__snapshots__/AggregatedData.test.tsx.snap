// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div>
  <div
    class="MuiBox-root MuiBox-root-1"
  >
    <table
      class="MuiTable-root"
    >
      <tbody
        class="MuiTableBody-root"
      >
        <tr
          class="MuiTableRow-root"
        >
          <td
            class="MuiTableCell-root MuiTableCell-body"
          >
            <strong>
              Average:
            </strong>
             
            1200
          </td>
          <td
            class="MuiTableCell-root MuiTableCell-body"
          >
            <strong>
              Close:
            </strong>
             
            1201
          </td>
          <td
            class="MuiTableCell-root MuiTableCell-body"
          >
            <strong>
              High:
            </strong>
             
            1202
          </td>
          <td
            class="MuiTableCell-root MuiTableCell-body"
          >
            <strong>
              Low:
            </strong>
             
            1203
          </td>
        </tr>
      </tbody>
    </table>
    <div
      class="MuiBox-root MuiBox-root-2"
    >
      <table
        cellpadding="0"
        class="MuiTable-root makeStyles-root-3"
      >
        <tbody
          class="MuiTableBody-root"
        >
          <tr
            class="MuiTableRow-root"
          >
            <td
              class="MuiTableCell-root MuiTableCell-body MuiTableCell-alignLeft"
              style="width: 33vw; padding: 3px 10px; color: rgb(229, 33, 14); background-color: rgba(229, 33, 14, 0.082);"
            >
              <div
                class="makeStyles-priceCell-4"
              >
                <p
                  class="MuiTypography-root MuiTypography-body1"
                >
                  20000
                </p>
              </div>
            </td>
          </tr>
          <tr
            class="MuiTableRow-root"
          >
            <td
              class="MuiTableCell-root MuiTableCell-body MuiTableCell-alignRight"
              style="width: 33vw; padding: 3px 10px; color: rgb(0, 146, 57); background-color: rgba(0, 146, 57, 0.082);"
            >
              <div
                class="makeStyles-priceCell-4"
              >
                <p
                  class="MuiTypography-root MuiTypography-body1"
                >
                  12000
                </p>
              </div>
            </td>
          </tr>
        </tbody>
      </table>
    </div>
  </div>
</div>
`;
